### ISSUE TITLE
SagePay V3 Protocol (Question not issue)

### ISSUE THREAD

**Issue by MrChriZ** [JeremySkinner/SagePayMvc#12]
Technically this isn't an issue.  I don't think there's a forum here though and I think it's of common interest.

Is SagePayMvc now tested and working with the SagePay V3 protocol?

Thanks
Chris.

**Comment by JeremySkinner** [comment]
Hi Chris

It should work just fine with v3.

The only difference is how signatures are generated, so you can no longer
validate signatures in your notification controller. There's no way to fix
this as sagepay no longer publish how the signatures are generated (and
their own integration kits no longer do signature validation)

Jeremy

**Comment by MrChriZ** [comment]
That's great to know. Thanks Jeremy.

**Comment by Cogiva** [comment]
The version I am using seems to not to play with V3 (I'm till getting emails from SagePay saying that we need to upgrade).  Has anything changed specifically for V3?

**Comment by MrChriZ** [comment]
Primarily it's jusy the version number sent through on each request.  There's a few minor behind the scenes changes.  You should be able to drop the latest build straight in.

**Comment by Cogiva** [comment]
Thanks. I did in the end, but I was doing some digging to figure it where that version number was. Found it in the end. Thanks for the help... SagePay help is not particularly helpful!

**Comment by rippo** [comment]
Hi where did you find this in the sage pay control panel?

**Comment by rippo** [comment]
Actually after a little digging found out if you run in simulator mode you get this response
"Incorrect VPSProtocol version.  The value should be 2.23, not 3.0"

Changing to Test mode works ok. Luckily I have a sage pay test account to play with

**Comment by JeremySkinner** [comment]
Yeah, SagePay's simulator is basically obsolete - it only supports 2.23. They haven't updated it for 3.0.
